lib/github/middleware.js

```diff
@@ -105,4 +106,55 @@ module.exports = function middleware(callback) {
       }
     }
   });
+}
+
+
+function workerMiddleware(callbackName) {
+  return withSentry(async (context) => {
+    enhanceOctokit(context.github, context.log);
+
+    newrelic.setControllerName(`github/events.${context.name}`);
+    let webhookEvent = context.name;
+    if (context.payload.action) {
+      webhookEvent = `${webhookEvent}.${context.payload.action}`;
+    }
+    newrelic.addCustomAttributes({
+      'Webhook ID': context.id,
+      'Webhook Event': webhookEvent,
+      Repository: context.payload.repository,
+    });
+
+    context.sentry.setExtra('GitHub Payload', {
+      event: context.name,
+      action: context.payload.action,
+      id: context.id,
+      repo: (context.payload.repository) ? context.repo() : undefined,
+    });
+
+    const gitHubInstallationId = context.payload.installation.id;
+    context.log = context.log.child({ gitHubInstallationId });
+
+    if (context.payload.sender.type === 'Bot') {
+      context.log.info({ noop: 'bot', botId: context.payload.sender.id, botLogin: context.payload.sender.login }, 'Halting futher execution since the sender is a bot');
+      return;
+    }
+
+    if (isFromIgnoredRepo(context.payload)) {
+      context.log.info({ noop: 'ignored_repo', installation_id: context.payload.installation.id, repository_id: context.payload.repository.id }, 'Halting futher execution since the repository is explicilty ignored');
+      return;
+    }
+
+    const count = await Subscription.count({ where: { gitHubInstallationId } });
+    if (count < 1) {
+      context.log.info({ noop: 'no_subscriptions' }, 'Halting futher execution since no subscriptions were found');
+      return;
+    }
+
+    await queues.webhook.add({ callbackName, context });
```

Comment:
instead of having two copies of this function, would it be possible to to the if env var here instead? Basically:
```
If use_bull_queue:
  await queues.webhook.add
else:
  original logic
```

Since all the logic is the same, it's just encapsulated in another file, you could call that other file for either. It makes me feel safer considering possible drift where one function gets updated but not the other, also means that all the original tests will be touching the code that will be running, no matter the flag.

Reply:
@tebriel can you expand a bit more on what you're thinking here and the motivation? The way we call these functions are pretty different. I did add an ENV switch but it's to use the inline vs async versions.

My approach here is to temporarily keep this not-DRY in a short term as we transition, then deleting the original `middleware` function.

Reply:
Yeah, so my thought was: we moved all of the logic from `middleware.js` after the subscription select, to another file. It's doing the same thing, but one has some extra decoration. If we could encapsulate that logic into a single function, then the original path could call it and the new path could also call it (but maybe with some pre-work). This way we make sure there weren't any typos/changes/missed scope variables/etc from the logic that gets moved into the new file. This is just the core business logic of the endpoint, so it seems like it should be the same for sync or bull-queue processing.

Reply:
@tebriel and I talked over zoom on this. Recapping what we talked about here.

I totally agree with him that copy/pasting code is not a good long term solution. I'm trying to figure out the simplest change (and trying to avoid any refactors if I can) to have production. My gut says that backgrounding these requests will lead to the Jira integration being more stable, but I'm not certain. And I'm open to being wrong (and adjusting).

I'm hoping that next week I'll get that confidence. And I'll also expand this to:

* branch creation
* commit pushes

Which lead our "time spent" in NewRelic traces.

I also want to put together a datadog monitor before this goes out so that I can keep tabs on the queue in case we are not working it off fast enough.

Ideally this original method will be deleted by the end of next week. I do have a calendar reminder for two weeks though for me to hold me accountable to that. I do not want this code to be duplicated long-term.

@tebriel feel free to mention anything I may have missed! Thanks for the chat! and the suggestion!